<launch>

  <!-- Arguments -->
  <arg name="bag_file" value="/home/bjoshi/GX010172.bag" />
  <arg name="image_dir" value="/home/bjoshi/colmap/test/images" />
  <arg name="traj_file" value="/home/bjoshi/ros_workspaces/svin_ws/src/SVIn/pose_graph/svin_results/svin_2023_07_13_20_54_34.txt" />
  <arg name="left_image_topic" value="/gopro/image_raw" />
  <arg name="config_file" value="$(find utils_ros)/config/gopro/gopro1.yaml" />

  <arg name="scale" value="1.0" />
  <arg name="compressed" value="true" />

  <arg name="skip_first_line" value="true" />

  <node name="extract_keyframes" pkg="utils_ros" type="write_kf_images" output="screen">

    <param name="bag_file" value="$(arg bag_file)" />
    <param name="image_dir" value="$(arg image_dir)" />
    <param name="traj_file" value="$(arg traj_file)" />
    <param name="left_image_topic" value="$(arg left_image_topic)" />
    <param name="config_file" value="$(arg config_file)" />

    <param name="scale" value="$(arg scale)" />
    <param name="compressed" value="$(arg compressed)" />

    <param name="skip_first_line" value="$(arg skip_first_line)" />

  </node>

</launch>